src/api.F90

```diff
@@ -87,31 +91,93 @@ end function openmc_cell_get_id
+    real(8) :: min_temp       ! min common-denominator avail temp
+    real(8) :: max_temp       ! max common-denominator avail temp
+    real(8) :: temp           ! actual temp we'll assign
+    logical :: outside_low    ! lower than available data
+    logical :: outside_high   ! higher than available data
+
+    outside_low = .FALSE.
+    outside_high = .FALSE.
 
     err = E_UNASSIGNED
-    if (index >= 1 .and. index <= size(cells)) then
-      associate (c => cells(index))
-        if (allocated(c % sqrtkT)) then
-          n = size(c % sqrtkT)
-          if (present(instance) .and. n > 1) then
-            if (instance >= 0 .and. instance < n) then
-              c % sqrtkT(instance + 1) = sqrt(K_BOLTZMANN * T)
+
+    ! error if the cell is filled with another universe
+    if (cells(index) % fill /= NONE) then
+      err = E_INVALID_FILL
+    else
+      ! find which material is associated with this cell
+      if (present(instance)) then
+        material_ID = cells(index) % material(instance)
+      else
+        material_ID = cells(index) % material(1)
+      end if
+
+      ! index of that material into the materials array
+      material_index = material_dict % get_key(material_ID)
+
+      ! number of nuclides associated with this material
+      num_nuclides = size(materials(material_ID) % nuclide)
+
+      nuclide_index = materials(material_ID) % nuclide(1)
+      min_temp = minval(nuclides(nuclide_index) % kTs)
+      max_temp = maxval(nuclides(nuclide_index) % kTs)
+
+      do j = 2, num_nuclides
+        nuclide_index = materials(material_ID) % nuclide(j)
+        min_temp = max(min_temp, minval(nuclides(nuclide_index) % kTs))
+        max_temp = min(max_temp, maxval(nuclides(nuclide_index) % kTs))
+      end do
+
+      ! adjust the temperature to be within bounds if necessary
+      if (K_BOLTZMANN * T < min_temp) then
+        outside_low = .TRUE.
+        temp = min_temp / K_BOLTZMANN
+      else if (K_BOLTZMANN * T > max_temp) then
+        outside_high = .TRUE.
+        temp = max_temp / K_BOLTZMANN
+      else
+        temp = T
+      end if
+
+      if (index >= 1 .and. index <= size(cells)) then
+        associate (c => cells(index))
+          if (allocated(c % sqrtkT)) then
+            n = size(c % sqrtkT)
+            if (present(instance) .and. n > 1) then
+              if (instance >= 0 .and. instance < n) then
+                c % sqrtkT(instance + 1) = sqrt(K_BOLTZMANN * temp)
+                err = 0
+              end if
+            else
+              c % sqrtkT(:) = sqrt(K_BOLTZMANN * temp)
               err = 0
             end if
-          else
-            c % sqrtkT(:) = sqrt(K_BOLTZMANN * T)
-            err = 0
           end if
+        end associate
+
+        ! assign error codes for outside of temperature bounds provided the
+        ! temperature was changed correctly. This needs to be done after
+        ! changing the temperature based on the logical structure above.
```

Comment:
Does using a warning that's printed from OpenMC not go against structuring OpenMC as a library (so that the calling code decides what to do given E_BELOW_MIN_BOUND/E_ABOVE_MAX_BOUND return values - either print it's own mooseWarning, throw its own mooseError, etc.)?